service/history/workflowExecutionContext.go

```diff
@@ -187,12 +187,13 @@ func (c *workflowExecutionContext) updateWorkflowExecution(transferTasks []persi
 			c.msBuilder.GetExecutionInfo().LastFirstEventID, c.msBuilder.GetExecutionInfo().NextEventID)
 	}
 
-	return c.updateHelper(nil, transferTasks, timerTasks, c.createReplicationTask, "", currentVersion, transactionID)
+	now := common.NewRealTimeSource().Now()
```

Comment:
Lets call the API UtcNow and have a timesource defined on workflowExecutionContext